### PR TITLE
Break down incoming events to publish separately

### PR DESCRIPTION
**Pull request by souradeep-das**
:clipboard: Partly https://github.com/omisego/elixir-omg/issues/1426

## Overview

This enables to break down and send separate events to Datadog, when events consumed from Ethereum event listeners come in bulk

## Changes

- Update `OMG.Watcher.DatadogEvent.ContractEventConsumer` and the Childchain sibling to iterate and send received events individually 
- Update tests to check breaking into separate events

## Testing
```
mix test test/omg_watcher/datadog_event/contract_event_consumer_test.exs
mix test test/omg_child_chain/datadog_event/contract_event_consumer_test.exs
```

### CODE REVIEW

File: apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex [hunk 1]
```diff
@@ -84,9 +84,11 @@ defmodule OMG.ChildChain.DatadogEvent.ContractEventConsumer do
 
     options = tags(aggregation_key, state.release, state.current_version, timestamp)
     title = "#{event_name}"
-    message = "#{inspect(Encode.make_it_readable!(data))} - Timestamp: #{timestamp}"
 
-    :ok = apply(state.publisher, :event, create_event_data(title, message, options))
+    Enum.each(data, fn ev ->
+      message = "[#{inspect(Encode.make_it_readable!(ev))}] - Timestamp: #{timestamp}"
+      :ok = apply(state.publisher, :event, create_event_data(title, message, options))
```
Comment (unnawut) [inline apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex:90]: The `title` is derived from `hd(data)`, so it's assuming the first event name is the same as the rest.

In this PR where we're already breaking down per event, should we derive it from each event instead?

File: apps/omg_child_chain/test/omg_child_chain/datadog_event/contract_event_consumer_test.exs [hunk 2]
```diff
@@ -41,22 +41,23 @@ defmodule OMG.ChildChain.DatadogEvent.ContractEventConsumerTest do
...
-  test "if a list of event message are put on omg bus is consumed by the event consumer and published on the publisher interface" do
+  test "if a list of events put on omg bus is consumed by the event consumer, broken down and published individually on the publisher interface" do
     topic_name = self() |> :erlang.pid_to_list() |> to_string()
     sig = "#{topic_name}(bytes32)"
     data = [%{event_signature: sig}, %{event_signature: sig}]
     {:root_chain, topic_name} |> OMG.Bus.Event.new(:data, data) |> OMG.Bus.direct_local_broadcast()
 
-    assert_receive {:event, _, _}
+    Enum.each(data, fn _ ->
+      assert_receive {:event, _, _}
```
Comment (unnawut) [inline apps/omg_child_chain/test/omg_child_chain/datadog_event/contract_event_consumer_test.exs:59]: Now that it expect 2 events, should we check that they're indeed the two events, and not one event reported twice?

File: apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex (hunk 1 above)
Comment (souradeep-das) [reply apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex:90]: True, that would make more sense. Is there any event other than the `BlockSubmitted` which could come in bulk?

File: apps/omg_child_chain/test/omg_child_chain/datadog_event/contract_event_consumer_test.exs (hunk 2 above)
Comment (souradeep-das) [reply apps/omg_child_chain/test/omg_child_chain/datadog_event/contract_event_consumer_test.exs:59]: Absolutely, i'm taking the message into equation then

File: apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex (hunk 1 above)
Comment (unnawut) [reply apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex:90]: I didn't look into specifics but I think `DepositCreated` and exit started events. Need double checking though, I havn't touched on them lately.

Comment (kevsul) [reply apps/omg_child_chain/lib/omg_child_chain/datadog_event/contract_event_consumer.ex:90]: And `ExitFinalized`, some others as well

**Pull request merged**